eth/state_accessor.go

```diff
@@ -30,73 +30,90 @@ import (
 	"github.com/ethereum/go-ethereum/trie"
 )
 
+// noopReleaser is returned in case there is no operation expected
+// for releasing state.
+var noopReleaser = func() {}
+
 // StateAtBlock retrieves the state database associated with a certain block.
 // If no state is locally available for the given block, a number of blocks
 // are attempted to be reexecuted to generate the desired state. The optional
-// base layer statedb can be passed then it's regarded as the statedb of the
+// base layer statedb can be provided which is regarded as the statedb of the
 // parent block.
+//
+// An additional release function will be returned if the requested state is
+// available. The release is expected be invoked when the state is used, though
+// it can be noop in some cases, otherwise the resources leaking can happen.
```

Comment:
```suggestion
// An additional release function will be returned if the requested state is
// available. Release is expected to be invoked when the returned state is no longer needed.
// Its purpose is to prevent resource leaking. Though it can be noop in some cases.
```